[FEA][new_main] + hipblas example with CU mask (1/2)

domcharrier opened the pull request:
Adds an example `hipblas_with_numpy_and_cu_mask.py` that demonstrates
how to create a stream with a CU mask and how to pass it do different
HIP and hipBLAS APIs.

## TODOs

* [x] Get code generator PR reviewed.
* [ ] Upload new ROCm 6.3.2 sources and bindings.
* [x] Test this PR. See internal build `#91`

domcharrier committed:
examples/0_*: + hipblas example with CU mask

Adds an example `hipblas_with_numpy_and_cu_mask.py` that
demonstrates how to create a stream with a CU mask
and how to pass it do different HIP and hipBLAS
APIs.

psamfass reviewed (approved):
LGTM, RTM when this has been tested successfully.

domcharrier committed:
_rend*_upd*_vers*.py:  git {branch --show-current->rev-parse ...}

Use `git rev-parse --abbrev-ref HEAD` instead of `git branch --show-current` as it is supported by older Git versions too.

domcharrier commented:
Superseded. New releases bundle code generation base branch code, docs, examples and generated bindings.
